src/board.py

```diff
@@ -0,0 +1,82 @@
+import math
+from random import randint
+from exception import OutOfBoardException
+from enum import IntEnum
+
+FENCE_SIZE = 8
+BASE_LINE_SIZE = 17
+
+
+class Item(IntEnum):
+    NO_FENCE = 0
+    FENCE = 1
+    SQUARE = 2
+    PAWN = 3
+
+
+class Direction(IntEnum):
+    NO = 0
+    HORIZONTALLY = 1
+    VERTICALLY = 2
+
+
+def build():
+    fences = [[Direction.NO for i in range(FENCE_SIZE)] for j in range(FENCE_SIZE)]
+    for i in range(10):
+        center_x = randint(0, FENCE_SIZE - 1)
+        center_y = randint(0, FENCE_SIZE - 1)
+        direction = randint(Direction.HORIZONTALLY, Direction.VERTICALLY)
+        fences[center_x][center_y] = direction
+    return fences
+
+
+def get_board(pawn, fences):
+    if (is_out_of_board(pawn)):
+        raise OutOfBoardException("The pawn is out of the board")
+    board = generate_default_board(pawn)
+    return add_fences(board, fences)
+
+
+def generate_default_board(pawn):
+    board = []
+    for y in range(BASE_LINE_SIZE):
+        line = []
+        for x in range(BASE_LINE_SIZE):
+            if pawn.x == x and pawn.y == y:
+                line.append(Item.PAWN)
+            elif is_odd(x) or is_odd(y):
+                line.append(Item.NO_FENCE)
+            else:
+                line.append(Item.SQUARE)
+        board.append(line)
+    return board
+
+
+def is_odd(number):
+    return number % 2 != 0
+
+
+def add_fences(board, fences):
+    new_board = board.copy()
```

Comment:
If you have a 2D array you should use deepcopy. Copy don't copy nested array. 
IE:
```python
>>> a = [[1, 2, 3, 4]]
>>> b = a.copy()
>>> b[0][0] = 2
>>> a
[[2, 2, 3, 4]]
```